ads/google/imaVideo.js

```diff
@@ -221,241 +195,154 @@ let consentState;
+  const attributeName = `data-${state}`;
+  if (active) {
+    root.setAttribute(attributeName, '');
+  } else {
+    root.removeAttribute(attributeName);
+  }
+}
+
+/**
+ * @param {Document} elementOrDoc
+ * @return {!Object<string, !Element>}
+ */
+function renderElements(elementOrDoc) {
+  const html = htmlFor(elementOrDoc);
+
+  // Elements annotated with `ref="name"` are referenced as `elements['name']`.
+  // They also have their `ref` copied as a classname, so they can be selected
+  // from CSS using that exact value.
+  const root = html`
+    <div class="fill">
+      <div class="fill">
+        <video
+          ref="video"
+          playsinline
+          controlslist="nodownload nofullscreen noremoteplayback"
+        >
+          <!-- Video children are later propagated from the iframe's name. -->
+        </video>
+      </div>
+
+      <div ref="adContainer" class="fill" hidden>
+        <!-- This subtree may be modified by the IMA SDK. -->
+      </div>
+
+      <div ref="controls" hidden>
+        <button ref="playButton"></button>
+
+        <div class="countdownWrapper">
+          <div ref="countdown">
+            <!-- Text content updated using data-ad-label onAdProgress(). -->
+          </div>
+        </div>
+
+        <div ref="time">
+          <!-- Text content must match format in updateTime(). -->
+          -:- / 0:00
+        </div>
+
+        <div ref="progress">
+          <div ref="progressLine"></div>
+          <div ref="progressMarker"></div>
+        </div>
+
+        <button ref="muteButton"></button>
+        <button ref="fullscreenButton"></button>
+      </div>
+
+      <button ref="overlayButton" class="fill"></button>
+    </div>
+  `;
+
+  const elements = htmlRefs(root);
+
+  // The root element cannot be referenced by `ref="root"`, so we insert it.
+  elements['root'] = root;
+
+  // For a smaller template, we copy each element's `ref` value as a classname.
+  for (const ref in elements) {
+    elements[ref].classList.add(ref);
+  }
+
+  // Adding SVGs separately because they require a different namespace, and the
+  // HTML template must be static.
+  const svg = svgFor(elementOrDoc);
+
+  elements['overlayButton'].appendChild(icons.play(svg));
+  elements['fullscreenButton'].appendChild(icons.fullscreen(svg));
+
+  // Buttons toggle SVGs by including two each, one is displayed at a time.
+  // See CSS selectors for buttons under .root[data-*].
+  const {'playButton': playButton, 'muteButton': muteButton} = elements;
+
+  playButton.appendChild(icons.play(svg));
+  playButton.appendChild(icons.pause(svg));
+
+  muteButton.appendChild(icons.volumeMax(svg));
+  muteButton.appendChild(icons.muted(svg));
+
+  return elements;
+}
+
 /**
  * @param {!Object} global
  * @param {!Object} data
  */
 export function imaVideo(global, data) {
+  insertCss(global.document.head, cssText);
+
   videoWidth = global./*OK*/ innerWidth;
   videoHeight = global./*OK*/ innerHeight;
```

Comment:
We were previously setting these constant sizes to elements we own. Since it always matched the window height, we now rely on the `fill` classname instead.

They're still passed to the IMA SDK on creation and the downstream `resize` event. (The latter can likely  be replaced with a local `ResizeObserver`, but that's orthogonal to this change.)